lldb/source/ValueObject/DILEval.cpp

```diff
@@ -1708,19 +1810,33 @@ Interpreter::Visit(const BinaryOpNode *node) {
                                                     "result");
     }
 
+    // Problem: cannot check the type of RHS without evaluating it anymore.
+    // If LHS is true, ignore rhs entirely without producing a potential error?
+    // Or evaluate both anyway to produce a correct error?
+    // LLDB produces error for RHS anyway:
+    // (lldb) expr true || s
```

Comment:
As mentioned elsewhere, we need to ignore the rhs and not produce a potential error, if LHS is true. Otherwise we might end up evaluating an RHS with a side effect when we shouldn't.

Reply:
Or rather "if evaulating the lhs allows us to skip evaluating the rhs".  An lhs that is evaluates to 'true' doesn't let us skip if the operator is a logical-AND. (But you knew that). :-)